[master] Make changes to remove SaltStack from tutorial and invalid deprecation notice

barbaricyawps opened the pull request:
### What does this PR do?

This PR updates the Salt in 10 minutes tutorial to remove "SaltStack" and adjust Tom's title per a request from Chunga.

It also resolves a customer request from Rob H to remove inaccurate version information from the Salt Cloud on Windows documentation.

I'm also adding the release notes file for 3007 since it's breaking local Nox builds of the docs.

### What issues does this PR fix or reference?
Fixes: https://jira.eng.vmware.com/browse/VRAE-50911

barbaricyawps committed:
Make changes requested by Chunga and Rob H

barbaricyawps committed:
Add release notes file for 3007 to allow nox builds

s0undt3ch commented on doc/topics/releases/3007.0.md:
Don't commit this file.

Any additions should be done in the template file, https://github.com/saltstack/salt/blob/master/doc/topics/releases/templates/3007.0.md.template

barbaricyawps replied:
Thanks, Pedro. I'll remove the file from the PR.

Would I still get a docs build failure without the md file, though? That's the problem we're trying to solve. The build fails because it wants this missing md file.

s0undt3ch replied:
> Thanks, Pedro. I'll remove the file from the PR.
> 
> Would I still get a docs build failure without the md file, though? That's the problem we're trying to solve. The build fails because it wants this missing md file.

The CI/CD pipeline takes care of creating the file

barbaricyawps replied:
Awesome. That takes care of my concerns, then. I'll remove the file soon and push up the changes.

barbaricyawps committed:
Add fixes requested by reviewers

barbaricyawps commented:
I've added changes requested by @whytewolf to clarify which versions need the impacket dependencies.

I've also removed the 3007.0.md requested by @s0undt3ch .

Would you mind taking a second look? Thanks!

barbaricyawps committed:
Further changes requested by reviewers
